### PR TITLE
Rework production palette audio feedback.

### LINKED ISSUE #16033: Training notification rate limit removes important audio feedback.
#15360 has generated some legitimate complaints for removing what is currently the only feedback for queuing a new production item.

There are a couple of routes for fixing this:

1) Revert #15360 and then defer the fix to some future implementation that implements audio channels and cancelling the old sound instead of ignoring the new one.
2) (my preferred option) enable the button click sounds on the production palette to provide direct audio feedback

IMO we could/should take 2 a step further and only play the training/building clip on the first item or un-pausing, relying on the standard button feedback for all later items.  This makes the audio response predictable, instead of changing based on the rate of player input. It also matches the original behaviour in RA2 / TS.

### PR DESCRIPTION
**Pull request by pchote**
Fixes #16033 by copying what RA2 does.

### CODE REVIEW

**Review by Smittytron — DISMISSED** [review]

**Review by TheChosenEvilOne — DISMISSED** [review]

**Comment by pchote** [comment]
Rebased, and replaced the `null`s in the `PlayNotification` calls with `World.LocalPlayer` for consistency with the rest of the file (it makes no difference in the Sound code).

**Review by ghost — APPROVED** [review]

File: OpenRA.Mods.Common/Widgets/ProductionPaletteWidget.cs [hunk 1]
```diff
@@ -267,14 +267,15 @@ bool HandleLeftClick(ProductionItem item, ProductionIcon icon, int handleCount,
 		{
 			if (PickUpCompletedBuildingIcon(icon, item))
 			{
-				Game.Sound.Play(SoundType.UI, TabClick);
+				Game.Sound.PlayNotification(World.Map.Rules, World.LocalPlayer, "Sounds", ClickSound, null);
```
Comment (teinarss) [inline OpenRA.Mods.Common/Widgets/ProductionPaletteWidget.cs]: Maybe convert the "Sounds" to a const and use it on all the places

Comment (pchote) [reply OpenRA.Mods.Common/Widgets/ProductionPaletteWidget.cs]: "Sounds" is used across 17 files, so would rather not do that in this PR

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/361e73959154bb7456393f53350ebc85c65b7dce

76b15d7e37dea3500761591fed4abc6124a533e5
cc2912b99357d400b1b5a08cf005d436a9c947d0

**Comment by netnazgul** [comment]
And there is a problem with "Unit ready" where it actually doesn't get voiced at all if another speech sound is ongoing, like "Building" or "On Hold", so you might miss the actual production of a unit.

"Construction complete" on the other hand does allow overlapping itself with production from base tab and defence tab.

**Pull request merged**